src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -169,6 +170,41 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         return tagSet;
     }
 
+    /**
+     * Parses a {@code String medicalHistory} into a {@code medicalHistory}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public static MedicalHistory parseMedicalHistory(String medicalHistory) throws ParseException {
+        requireNonNull(medicalHistory);
+        String trimmedMedicalHistory = medicalHistory.trim();
+        if (!MedicalHistory.isValidMedicalHistory(medicalHistory)) {
+            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        }
+        return new MedicalHistory(trimmedMedicalHistory);
+    }
+
+    /**
+     * Parses a collection of medical history strings and returns a Set of MedicalHistory objects.
+     * @param medicalHistoryStrings The collection of medical history strings.
+     * @return A Set of MedicalHistory.
+     * @throws ParseException if any medical history format is invalid.
+     */
+    public static Set<MedicalHistory> parseMedicalHistories(Collection<String> medicalHistoryStrings)
+            throws ParseException {
+        requireNonNull(medicalHistoryStrings);
+
+        final Set<MedicalHistory> medicalHistories = new HashSet<>();
+        for (String medicalHistoryString : medicalHistoryStrings) {
+            if (medicalHistoryString.isBlank()) {
+                throw new ParseException("Medical history cannot be empty.");
+            }
+            medicalHistories.add(new MedicalHistory(medicalHistoryString));
+        }
+        return medicalHistories;
+    }
+
```

Comment:
`parseMedical` and `parseMedicalHistory` does the exact same thing! We can compress these 2 methods into 1 singular one.

Reply:
Will KIV when refactoring the code base.